lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -6,7 +6,14 @@ defmodule Mix.Compilers.Elixir do
   import Record
 
   defrecordp :module, [:module, :kind, :source, :beam, :binary]
-  defrecordp :source, [:source, compile: [], runtime: [], external: []]
+  defrecordp :source, [
+    source: nil,
+    compile: [],
+    runtime: [],
+    compile_dispatches: [],
+    runtime_dispatches: [],
```

Comment:
Do we care if the dispatch was compile time or not? Maybe we should just put them all into the same bucket called `compile_dispatches`? In fact, the only thing I can think of is that we don't need to check `compile_dispatches` because if it happened at compile time, it certainly worked. :) Let's talk on IRC later when you are back about the next steps.